node/src/actors/chain_manager/handlers.rs

```diff
@@ -403,6 +404,13 @@ impl Handler<AddBlocks> for ChainManager {
                         let (batch_succeeded, num_processed_blocks) =
                             self.process_first_batch(ctx, &sync_target, &blocks);
                         if !batch_succeeded {
+                            // We receive an invalid batch of blocks. So we need to throw away our
+                            // outbound peers in order to find new ones that can give us the blocks
+                            // consolidated by the network
+                            self.drop_all_outbounds();
+                            // Ice the invalid blocks' batch sender
+                            self.ice_peer(sender);
+
```

Comment:
Do you think it could be worth refactoring this code fragment into a `drop_all_outbounds_and_ice_batch_sender` function or method that operates on `self` / `act` and `sender` so as to avoid code repetition?

Reply:
The whole `AddBlocks` handler needs a refactor...

Reply:
the idea was to have some functions that could be used in another parts of the code, so that's the reason to do not a so specific function